File: diagonal_difference/diagonal_difference.go
package main
import (
    "fmt"
    "io"
    "log"
    "math"
)

func main() {
    var values int
    var temp [] int
    fmt.Scanf("%v/n", &values)
    var arr [][] int
    for i := 0; i < values; i++ {
        temp = readLine(values)
        arr = append(arr, temp)
    }
    var diagA int
    var diagB int
    for j :=0; j < values; j++ {
        diagA += arr[j][j];
        diagB += arr[j][values - 1 - j];
    }
    fmt.Println(math.Abs(float64(diagA - diagB)))
}


func readLine(x int) [] int {
    var array [] int
    var d int
    for k :=0; k < x; k++ {
        _, err := fmt.Scan(&d)
        if err != nil {
            if err != io.EOF {
                log.Fatal(err)
            }
            break
        }
        array = append(array, d)
    }
    return array
}

File: diagonal_difference/diagonal_difference.js
function main() {
  var n = parseInt(readLine());
  var a = [];
  for (a_i = 0; a_i < n; a_i++) {
    a[a_i] = readLine().split(' ');
    a[a_i] = a[a_i].map(Number);
  }
  var diagA = 0;
  var diagB = 0;

  for (var i = 0; i < n; i++) {
    diagA += a[i][i];
    diagB += a[i][n - 1 - i];
  }
  console.log(Math.abs(diagA - diagB));
}
